### PR TITLE
Handle better uninstall/install errors

### PR DESCRIPTION
**Pull request by frankrousseau**
Two few information was displayed in case of errors. This PR attempts to improve things by displaying more details about errors and displaying "guess" messages about what's wrong.

### CODE REVIEW

**Comment by frankrousseau** [comment]
@poupotte Could you review my PR?

**Comment by jsilvestre** [comment]
Since you inherit the TR job for the platform @aenario, could you please review this PR? Thanks!

**Comment by aenario** [comment]
on it

File: lib/main.coffee [hunk 1]
```diff
@@ -382,22 +410,39 @@ program
             path = "api/applications/install"
             homeClient.post path, manifest, (err, res, body) ->
                 if err or body.error
-                    isIndexOf = body.message.indexOf('Not Found')
-                    if body?.message? and  isIndexOf isnt -1
-                        err = """
+                    if err.code is 'ECONNREFUSED'
```
Comment (aenario) [inline lib/main.coffee]: break if there is an body.error but no err

File: lib/main.coffee [hunk 2]
```diff
@@ -294,13 +319,16 @@ installApp = (app, callback) ->
...
-            if err
-                callback err
-            else if body.error
-                callback body.error
-            else
-                callback()
+        if err
+            callback err
+        else if body?.error
+            callback body.error
```
Comment (aenario) [inline lib/main.coffee]: should not callback String

**Comment by aenario** [comment]
I made a couple comment.
cozy-monitor really needs some refactoring.

otherwise manually tested and few command and works

**Comment by jsilvestre** [comment]
> cozy-monitor really needs some refactoring.

I'm going to work with @poupotte on a backlog for that.

**Comment by frankrousseau** [comment]
Please, focus on your current sprint. Build a backlog once the Cozy Monitor is part of the selected sprints.

**Comment by frankrousseau** [comment]
I made two fixes requested by @aenario. Let me know I have to change something else.

**Comment by aenario** [comment]
published with tag v1.2.9

**Pull request merged**